src/libraries/System.Private.CoreLib/src/System/Runtime/JitInfo.cs

```diff
@@ -0,0 +1,32 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+namespace System.Runtime
+{
+    public static partial class JitInfo
+    {
+        private const long TicksPerMillisecond = 10000;
+        private const long TicksPerSecond = TicksPerMillisecond * 1000;
+
+        // "Frequency" stores the frequency of the high-resolution performance counter,
+        // if one exists. Otherwise it will store TicksPerSecond.
+        // The frequency cannot change while the system is running,
+        // so we only need to initialize it once.
+        public static readonly long Frequency = System.Diagnostics.Stopwatch.QueryPerformanceFrequency();
+
+        // pre calculating the tick frequency for quickly converting from QPC ticks to DateTime ticks
+        private static readonly double s_tickFrequency = (double)TicksPerSecond / Frequency;
+
+        /// <summary>
+        /// Get the amount of time the JIT Compiler has spent compiling methods. If <paramref name="currentThread"/> is true,
+        /// then this value is scoped to the current thread, otherwise, this is a global value.
+        /// </summary>
+        /// <param name="currentThread">Whether the returned value should be specific to the current thread. Default: false</param>
+        /// <returns>The amount of time the JIT Compiler has spent compiling methods.</returns>
+        public static TimeSpan GetCompilationTime(bool currentThread = false)
+        {
+            // See System.Diagnostics.Stopwatch.GetElapsedDateTimeTicks()
+            return TimeSpan.FromTicks((long)(GetCompilationTimeInTicks(currentThread) * s_tickFrequency));
```

Comment:
This looks good for CoreCLR, but I believe Mono needs a separate implementation, since the value it returns actually uses 100 nanosecond ticks: https://github.com/dotnet/runtime/blob/41e6601c5f9458fa8c5dea25ee78cbf121aa322f/src/mono/mono/mini/mini.c#L4069-L4072

Reply:
My impression was that the `QueryPerformancFrequency` calls on Mono should account for that, but I'll double check to be sure.

Reply:
The problem is that Mono already does the conversion for you, so you do not need to do it again:
https://github.com/dotnet/runtime/blob/57bfe474518ab5b7cfe6bf7424a79ce3af9d6657/src/mono/mono/utils/mono-time.c#L84